add examples for docker inspect

k3ck3c opened the pull request:
We can get the whole list of environment variables with

docker inspect -f '{{ index (index .Config.Env) }}' 797
[DISPLAY=:0 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin LANG=fr_FR.UTF-8 LANGUAGE=fr_FR:en LC_ALL=fr_FR.UTF-8 DEBIAN_FRONTEND=noninteractive HOME=/home/gg WINEARCH=win32 WINEPREFIX=/home/gg/.wine_captvty]

We can get the first environment variable with

docker inspect -f '{{ index (index .Config.Env) 0 }}' 797 
DISPLAY=:0

or the second
docker inspect -f '{{ index (index .Config.Env) 1 }}' 797 
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

We can go from a more general to a more precise display

docker inspect --format '{{ (.NetworkSettings.Ports) }}' 87c
map[8000/tcp:[map[HostIp:0.0.0.0 HostPort:49153]]]

if we only want what follows the string "8000/tcp", we can do 

docker inspect --format '{{ ((index .NetworkSettings.Ports "8000/tcp") 0) }}' 87c
[map[HostIp:0.0.0.0 HostPort:49153]]

docker inspect --format '{{ index (index .NetworkSettings.Ports "8000/tcp") 0 }}' 87c
map[HostIp:0.0.0.0 HostPort:49153]

If we only want what follows HostIp
docker inspect --format '{{ (index (index .NetworkSettings.Ports "8000/tcp") 0).HostIp }}' 87c
0.0.0.0

If we only want what follows HostPort
docker inspect --format '{{ (index (index .NetworkSettings.Ports "8000/tcp") 0).HostPort }}' 87c
49153

HuKeping committed:
Fix inconsistent of remote API and document

Signed-off-by: Hu Keping <hukeping@huawei.com>

Commit:
Fixes #13031 - Check for upstart or init is not detecting properly

This will now properly check whether /etc/init.d/docker or service docker is
invoking the script and respond to the user accordingly.

Signed-off-by: Steven Richards <steven@axiomzen.co>

Commit:
Refactor TLS code with a new `tlsconfig` package

This patch creates a new `tlsconfig` package to handle creation of
secure-enough TLS configurations for clients and servers.

The package was created by refactoring TLS code in the client and the
daemon. After this patch, it is expected that all code creating TLS
configurations use this `tlsconfig` package for greater security,
consistency and readability.

On the server side, this fixes a bug where --tlsverify was not taken
into account. Now, if specified, it will require the client to
authenticate.

Signed-off-by: Tibor Vass <tibor@docker.com>

ahmetb committed:
Allow duration strings as --since/--until

Fixes #13107. This change enables Go duration strings
computed relative to the client machine’s time to be used
as input parameters to `docker events --since/--until`
and `docker logs --since` arguments.

Added unit tests for pkg/timeutils.GetTimestamp as well.

Signed-off-by: Ahmet Alp Balkan <ahmetalpbalkan@gmail.com>

Commit:
Add a "Created" state for new containers that haven't been run yet

Closes #13353

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Add syslog-address log-opt

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:docker/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~1733
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

This will update the existing PR, so **DO NOT** open a new one.

cpuguy83 commented:
Seems like you had an issue with a rebase or something.

Commit:
Fix regression in stats API endpoint where stream query param default is true

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
Remove PortSpecs from Config

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
Cleanup container LogEvent calls

Move some calls to container.LogEvent down lower so that there's
less of a chance of them being missed. Also add a few more events
that appear to have been missed.

Added testcases for new events: commit, copy, resize, attach, rename, top

Signed-off-by: Doug Davis <dug@us.ibm.com>

mrjana committed:
Vendoring in libnetwork 2da2dc055de5a474c8540871ad88a48213b0994f

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

moysesb committed:
Modified progress reader to close silently when read is complete.

Closes #13432

Signed-off-by: Moysés Borges <moysesb@gmail.com>

dwlf committed:
doc cli lint.

- 79 char line length (non-command).
- Consistent single space between sentences.
- Replace start of line tabs with spaces.
- Remove trailing white space.
- Consistent "**Note:**" style.
  https://docs.docker.com/project/doc-style/#notes
- Replace the ``` sections with indented code.

Signed-off-by: Lloyd Dewolf <foolswisdom@gmail.com>

cpuguy83 committed:
Use SELinux labels for volumes

Fixes a regression from the volumes refactor where the vfs graphdriver
was setting labels for volumes to `s0` so that they can both be written
to by the container and shared with other containers.
When moving away from vfs this was never re-introduced.
Since this needs to happen regardless of volume driver, this is
implemented outside of the driver.

Fixes issue where `z` and `Z` labels are not set for bind-mounts.

Don't lock while creating volumes

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Merge pull request #12572 from jfrazelle/selinux-labels-carry

Modify volume mounts SELinux labels on the fly based on :Z or :z

rscothern committed:
Restore the stripped registry version number

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

Commit:
Updating after the pr went through
Updating with comments.

Signed-off-by: Mary Anthony <mary@docker.com>

rscothern committed:
Review feedback:
    - Match verbiage with other output
    - Remove dead code and clearer flow

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

calavera committed:
Merge pull request #13496 from Microsoft/10662-windefaults

Windows: Set default exec driver to windows

tianon committed:
Merge pull request #13477 from kvasdopil/freebsd-build

Make hack/make.sh work on FreeBSD

Commit:
Merge pull request #13518 from mrjana/clone

Vendoring in libnetwork 2da2dc055de5a474c8540871ad88a48213b0994f

tiborvass committed:
Merge pull request #13374 from RichardScothern/v2-mirror

V2 mirror support

Commit:
Windows: Undo 908db518 for Windows daemon

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Fix error when trying to delete an image due to a bad container

I ran into a situation where I was trying:
	`docker rmi busybox`
and it kept failing saying:
    `could not find image: Prefix can't be empty`

While I have no idea how I got into this situation, it turns out this is
error message is from `daemon.canDeleteImage()`. In that func we loop over
all containers checking to see if they're using the image we're trying to
delete.  In my case though, I had a container with no ImageID. So the code
would die tryig to find that image (hence the "Prefix can't be empty" err).
This would stop all processing despite the fact that the container we're
checking had nothing to do with 'busybox'.

My change logs the bad situation in the logs and then skips that container.
There's no reason to fail all `docker rmi ...` calls just because of one
bad container.

Will continue to try to figure out how I got a container w/o an ImageID
but as of now I have no idea, I didn't do anything but normal docker cli
commands.

Signed-off-by: Doug Davis <dug@us.ibm.com>

jessfraz committed:
bump api version to 1.20

Signed-off-by: Jessica Frazelle <princess@docker.com>

jessfraz committed:
fix release script

Signed-off-by: Jessica Frazelle <princess@docker.com>

tianon committed:
Merge pull request #13532 from jfrazelle/fix-fix-release-script

fix release script

Mic92 committed:
remove redundant mount/unmount calls on commit

daemon.Diff already implements mounting for naivegraphdriver and
aufs which does diffing on its owns does not need the container to be mounted.
So new filesystem driver should mount filesystems on their own if it is needed
to implement Diff(). This issue was reported by @kvasdopil while working on a
freebsd port, because freebsd does not allow mount an already mounted
filesystem. Also it saves some cycles for other operating systems as well.

Signed-off-by: Jörg Thalheim <joerg@higgsboson.tk>

hqhq committed:
Skip test if not have Cpu quota or Cpu period

Closes: #13522

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

thaJeztah committed:
Merge pull request #13515 from lloydde/doc-cli-lint

doc cli lint.

tianon committed:
Add fedora:22 to our rpm targets

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

tianon committed:
Add Ubuntu Wily (15.10) as a build-deb target

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

coolljt0725 committed:
Clean up: remove redundant '\n' in Create

Signed-off-by: Lei Jitang <leijitang@huawei.com>

cpuguy83 committed:
Merge pull request #13541 from coolljt0725/remove_reduntant_enter_create

Clean up: remove redundant '\n' in Create

albers committed:
Use http for PGP key download

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
Update bash completion for 1.7.0

Signed-off-by: Harald Albers <github@albersweb.de>

kvasdopil committed:
zfs magicnumber check on freebsd is fixed

Signed-off-by: Alexey Guskov <lexag@mail.ru>

cpuguy83 committed:
Merge pull request #13535 from coolljt0725/fix_automatically_publish_without_publish

Fix automatically publish ports without --publish-all or --publish

cpuguy83 committed:
Merge pull request #13055 from Microsoft/10662-internalsfixperms

Windows: Fixup builder/internals.go

cpuguy83 committed:
Merge pull request #13269 from Microsoft/10662-statsrefactor

Windows: refactor stats

calavera committed:
Merge pull request #13538 from tianon/rpm-fedora-22

Add fedora:22 to our rpm targets

cpuguy83 committed:
Merge pull request #13259 from Microsoft/10662-configbridge

Windows: factor out bridgeConfig from server+config

LK4D4 committed:
Merge pull request #13548 from kvasdopil/zfs_magicnum

ZFS filesystem magic number check is fixed on FreeBSD

calavera committed:
Merge pull request #13039 from stevenbrichards/13031-Upstart

Fix check for upstart not detecting properly

calavera committed:
Merge pull request #13512 from moysesb/progressreader_close_silently

Close progressreader silently when read is complete.

Commit:
Merge pull request #13529 from hqhq/hq_skip_cgroup_test

Skip test if not have Cpu quota or Cpu period

calavera committed:
Merge pull request #13551 from albers/pgp-over-http

Use http for PGP key download

calavera committed:
Merge pull request #13539 from tianon/deb-ubuntu-wily

Add Ubuntu Wily (15.10) as a build-deb target

calavera committed:
Merge pull request #13428 from duglin/niceHelp2

Nice help2

calavera committed:
Merge pull request #13416 from Microsoft/10662-fixintegrationtests

Windows: Pass temp environment through

LK4D4 committed:
Merge pull request #13430 from runcom/fix-race-modify-request

Fix race in httpsRequestModifier.ModifyRequest when writing tlsConfig

ewindisch committed:
Expand unshare test to include privileged test

This ensures that AppArmor, not other mechanisms used
by Docker or the kernel is restricting the mount.

Signed-off-by: Eric Windisch <eric@windisch.us>

calavera committed:
Add comment about ulimit format.

Fixes #12116

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 committed:
Treat systemd listeners as all other

Fix #13549

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Windows: Factor out stat collector

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Add note about overlay not being production ready

Add a paragraph in cli.md mentioning that overlay is not a production
ready graphdriver.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Fixes to the 1.19 version
updating with changes to this instant
Signed-off-by: Mary Anthony <mary@docker.com>

calavera committed:
Mount bind volumes coming from the old volumes configuration.

Signed-off-by: David Calavera <david.calavera@gmail.com>

burke committed:
Use bufio.Reader instead of bufio.Scanner for logger.Copier

When using a scanner, log lines over 64K will crash the Copier with
bufio.ErrTooLong. Subsequently, the ioutils.bufReader will grow without
bound as the logs are no longer being flushed to disk.

Signed-off-by: Burke Libbey <burke.libbey@shopify.com>

Commit:
Merge pull request #13559 from LK4D4/fix_systemd_listen

Treat systemd listeners as all other

jessfraz committed:
skip test on lxc

Signed-off-by: Jessica Frazelle <princess@docker.com>

draghuram committed:
Make the version mismatch message more explicit.

Signed-off-by: Raghuram Devarakonda <draghuram@gmail.com>

Commit:
Merge pull request #13562 from moxiegirl/fixes-to-119-api

Fixes to the 1.19 version Thank you @thaJeztah and @jfrazelle!

calavera committed:
Merge pull request #13558 from calavera/doc_ulimit

Add comment about ulimit format.

Commit:
Merge pull request #13530 from jfrazelle/bump-api-version

bump api version to 1.20

Commit:
Merge pull request #13566 from jfrazelle/fix-lxc-tests

skip test on lxc

jessfraz committed:
fix experimental version and release script

add api version experimental

Signed-off-by: Jessica Frazelle <princess@docker.com>

anchal-agrawal committed:
Fixing Mac Boot2Docker installation doc

Signed-off-by: Anchal Agrawal <aagrawa4@illinois.edu>

WeiZhang555 committed:
bug fix: close http response body no longer in use

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

WeiZhang555 committed:
return error when failed to read http response body

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

stevvooe committed:
Properly verify manifests and layer digests on pull

To ensure manifest integrity when pulling by digest, this changeset ensures
that not only the remote digest provided by the registry is verified but also
that the digest provided on the command line is checked, as well. If this check
fails, the pull is cancelled as with an error. Inspection also should that
while layers were being verified against their digests, the error was being
treated as tech preview image signing verification error. This, in fact, is not
a tech preview and opens up the docker daemon to man in the middle attacks that
can be avoided with the v2 registry protocol.

As a matter of cleanliness, the digest package from the distribution project
has been updated to latest version. There were some recent improvements in the
digest package.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

s7v7nislands committed:
update lxc url

Signed-off-by: XiaoBing Jiang <s7v7nislands@gmail.com>

vdemeester committed:
Add test coverage to pkg/timeutils

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
Merge pull request #12773 from HuKeping/update-docs

Fix inconsistent of remote API and document

kvasdopil committed:
avoid 88-chars mountpoint length limit on freebsd

Signed-off-by: Alexey Guskov <lexag@mail.ru>

duglin committed:
Merge pull request #13583 from vdemeester/pkg-timeutils-test-coverage

Add test coverage to pkg/timeutils

vdemeester committed:
Add test coverage for pkg/stringutils

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 committed:
Merge pull request #13563 from calavera/fix_bind_mount_regression

Mount bind volumes coming from the old volumes configuration.

davidrjenni committed:
Fix issue #10184.

Merge user specified devices correctly with default devices.
Otherwise the user specified devices end up without permissions.

Signed-off-by: David R. Jenni <david.r.jenni@gmail.com>

Commit:
Merge pull request #13574 from anchal-agrawal/13345-fix-mac-install-docs

Fixing Mac Boot2Docker installation doc  Your fix has left the building.

tianon committed:
Merge pull request #13585 from albers/completion-1.7

Update bash completion for 1.7.0

calavera committed:
Merge pull request #13392 from runcom/syslog-connection-url-log-opt

Add syslog-address log-opt

LK4D4 committed:
Merge pull request #13482 from boucher/criu-libcontainer-update

Update vendored libcontainer

sanketsaurav committed:
Proposing Bose, Raman and Ramanujan

Signed-off-by: Sanket Saurav <sanketsaurav@gmail.com>

jessfraz committed:
fix bug with rmi multiple tag

Signed-off-by: Jessica Frazelle <princess@docker.com>

calavera committed:
Do not require cgroups capabilities on windows to run the integration tests.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Merge pull request #13581 from s7v7nislands/fix_lxc_url

update lxc url

tianon committed:
Finally add precise/12.04 as a build-deb target

Ubuntu Precise has a number of warts that made it non-trivial to add initially, but I've managed to work through some of them and come up with a working build.  Two important parts to note are that it has neither the `btrfs` nor the `devicemapper` graphdriver backends since `btrfs-tools` and `libdevmapper-dev` in the precise repositories are too ancient for them to even compile.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

calavera committed:
Merge pull request #13598 from calavera/fix_windows_build

Do not require cgroups capabilities on windows to run the integration tests.

tianon committed:
Make "DEST" a make.sh construct instead of ad-hoc

Using "DEST" for our build artifacts inside individual bundlescripts was already well-established convention, but this officializes it by having `make.sh` itself set the variable and create the directory, also handling CYGWIN oddities in a single central place (instead of letting them spread outward from `hack/make/binary` like was definitely on their roadmap, whether they knew it or not; sneaky oddities).

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

LK4D4 committed:
Merge pull request #13255 from vdemeester/11588-pkg-pools-test-coverage

Add test coverage for pkg/pools (#11588)

LK4D4 committed:
Merge pull request #13564 from burke/fix-memory-leak

Use bufio.Reader instead of bufio.Scanner for logger.Copier

LK4D4 committed:
Merge pull request #13589 from vdemeester/pkg-stringutils-test-coverage

Add test coverage for pkg/stringutils

tianon committed:
Add embedded shell script highlight to vim syntax

This highlights `RUN`, `CMD`, and `ENTRYPOINT` lines using shell highlighting.  It doesn't bother detecting the JSON forms, but that's OK because JSON arrays highlight pretty reasonably with shell highlights. :)

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

tianon committed:
Adjust vim ftdetect to match "Dockerfile", "dockerfile", and "Dockerfile.*" which are all reasonably safe to assume "this is a Dockerfile"

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

estesp committed:
Merge pull request #13578 from WeiZhang555/errRet

return error when failed to read http response body

estesp committed:
Merge pull request #13577 from WeiZhang555/httpClose

bug fix: close http response body no longer in use

Commit:
Merge pull request #13527 from duglin/FixImagesPrefixNotFound

Fix error when trying to delete an image due to a bad container

jfrazelle commented:
there is something wrong with your branch i think you need to rebase

Commit:
Merge pull request #13605 from tianon/vim-dockerfile

Adjust vim ftdetect to match "Dockerfile", "dockerfile", and "Dockerfile.*" which are all reasonably safe to assume "this is a Dockerfile"

Commit:
Merge pull request #13604 from tianon/vim-embedded-shell

Add embedded shell script highlight to vim syntax

stevvooe committed:
Attempt to retain tagging behavior

Signed-off-by: Stephen J Day <stephen.day@docker.com>

calavera committed:
Merge pull request #13446 from runcom/remove-PortSpecs

Remove PortSpecs from Config

Commit:
Merge pull request #13601 from tianon/precise

Finally add precise/12.04 as a build-deb target

runcom committed:
Merge pull request #13567 from draghuram/apiver

Make the version mismatch message more explicit.

calavera committed:
Merge pull request #13425 from runcom/13421-stats-default-stream

Fix regression in stats API endpoint

calavera committed:
Merge pull request #13596 from jfrazelle/11414-fix-once-and-for-all

fix bug with rmi multiple tag

Commit:
Revert api doc for 1.19 about PortSpecs removal

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
Merge pull request #13606 from runcom/revert-doc-in-api-1-19

Revert api doc for 1.19 about PortSpecs removal

thaJeztah committed:
Merge pull request #13561 from icecrime/warn_overlay_docs

Add note about overlay not being production ready

stevvooe committed:
Add tests for loadManifest digest verification

Signed-off-by: Stephen J Day <stephen.day@docker.com>

calavera committed:
Merge pull request #12400 from coolljt0725/kill_all_containers_on_daemon_shutdown

Ensure all the running containers are killed on daemon shutdown

vieux committed:
no not print empty keys in docker info

Signed-off-by: Victor Vieux <victorvieux@gmail.com>

Commit:
Allow .dockerignore to ignore everything

Change CLI error msg because it was too specific and didn't make sense
when there were errors not related to inaccessible files.

Removed some log.Error() calls since they're not really errors we should
log. Returning the error will be enough.

Closes: #13417

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Windows: Refix server_windows to match linux

Signed-off-by: John Howard <jhoward@microsoft.com>

vdemeester committed:
Update api documentation to add labels in commit

It is already possible to set labels at commit when using the API. But
it is not present in the API documentation. Added an integration test
too, to validate this work (and will be in the future).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
Premature newline insertions

Noticed that there were a few newlines in the middle of a \`...\` blocks that
made the text look funny when displayed in HTML.

Signed-off-by: Doug Davis <dug@us.ibm.com>

konstruktoid committed:
aes256 and 4096 bits
squashing down to a single commit and change back to cert.pem

Signed-off-by: Thomas Sjögren <konstruktoid@users.noreply.github.com>

jamtur01 committed:
Merge pull request #13612 from duglin/DocsNewLine

Premature newline insertions

thaJeztah committed:
Merge pull request #13580 from vdemeester/update-api-commit

Update api documentation to add labels in commit

coolljt0725 committed:
Add docker stats --no-stream show cpu usage

Signed-off-by: Lei Jitang <leijitang@huawei.com>

thaJeztah committed:
Merge pull request #13367 from duglin/Issue13353

Add a "Created" state for new containers that haven't been run yet

gesellix committed:
indicate HostConfig to be used for "create container"

Closes #13614

Signed-off-by: Tobias Gesellchen <tobias@gesellix.de>

jamtur01 committed:
Merge pull request #13616 from konstruktoid/docs_openssl

Docs: sha256 and 4096 when using openssl

jfrazelle committed:
Merge pull request #13603 from tianon/consistent-dest

Make "DEST" a make.sh construct instead of ad-hoc

Commit:
Merge pull request #13460 from moxiegirl/add-created-state-diagram

Updating event diagram with the created state. Cool, thank you.

jogo committed:
Fix spelling mistakes in documentation

While reading  some of the docs I noticed a few errors, so I ran
misspellings (https://pypi.python.org/pypi/misspellings) on markdown files

Signed-off-by: Joe Gordon <joe.gordon0@gmail.com>

SvenDowideit committed:
Merge pull request #13622 from jogo/spelling

Fix spelling mistakes in documentation

dalanlan committed:
fix typo.
Signed-off-by: He Simei <hesimei@zju.edu.cn>

cpuguy83 committed:
Merge pull request #13390 from mrjana/cnm_integ

Do not attempt releasing network when not attached to any network

thaJeztah committed:
Merge pull request #13615 from gesellix/docs-container-create-start

indicate HostConfig to be used for "create container"

Commit:
Merge pull request #13523 from moxiegirl/carry-12572-fix

Updating after the pr went through

estesp committed:
Merge pull request #13568 from jfrazelle/fix-release-script-for-experimental

fix release script for experimental

calavera committed:
Merge pull request #13464 from davidrjenni/10184-fix-auto-created-devices

Fixes issue #10184.

cpuguy83 committed:
Merge pull request #13320 from coolljt0725/add_stats_no_stream_show_cpu_usage

Add docker stats --no-stream show cpu usage

LK4D4 committed:
Remove useless debug message

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

cpuguy83 committed:
Merge pull request #13637 from LK4D4/remove_debug

Remove useless debug message

LK4D4 committed:
Merge pull request #13560 from Microsoft/10662-statscollector

Windows: Factor out stat collector

tonistiigi committed:
Fix breakouts from git root during build

Signed-off-by: Tõnis Tiigi <tonistiigi@gmail.com>

Commit:
Add @calavera to maintainers

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Windows: Implement ReadMemInfo()

Signed-off-by: John Howard <jhoward@microsoft.com>

estesp committed:
Merge pull request #13557 from ewindisch/apparmor-unshare-priv

Expand unshare test to include privileged test

stevvooe committed:
Break down loadManifest function into constituent parts

Signed-off-by: Stephen J Day <stephen.day@docker.com>

calavera committed:
Merge pull request #13639 from tonistiigi/fix-git-escape

Make sure git remote subdirs don't escape git root

LK4D4 committed:
Fix race condition in registry/session

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Fix - Download button does not download

Fix - Download button does not download

Users got confused when Kitematic is not being downloaded after clicking on the download button, and instead got brought to a screen to teach them how to download Kitematic.

The additional step caused a significant drop in Kitematic downloads. This pull request fixes the issue, to allow users to directly download Kitematic after clicking on the Download button. 

Signed-off-by: Michael Chiang <michael.chiang@docker.com>

calavera committed:
Fix stats flaky test.

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 committed:
Merge pull request #13443 from duglin/CleanupEvents

Cleanup container LogEvent calls

LK4D4 committed:
Merge pull request #13638 from icecrime/maintainers_add_calavera

Add @calavera to maintainers

jfrazelle committed:
Merge pull request #13643 from LK4D4/fix_registry_race

Fix race condition in registry/session

Commit:
registry: debugTransport should print with testing.T.Log

It should not print to STDOUT so that it only prints the debugTransport
output if there was an error in one of the registry tests.

Signed-off-by: Tibor Vass <tibor@docker.com>

jvgogh committed:
Upon HTTP 302 redirect do not include "Authorization" header on 'untrusted' registries.

Refactoring in Docker 1.7 changed the behavior to add this header where as Docker <= 1.6 wouldn't emit this Header on a HTTP 302 redirect.

This closes #13649

Signed-off-by: Jeffrey van Gogh <jvg@google.com>

rscothern committed:
If no endpoint could be established with the given mirror configuration,
fallback to pulling from the hub as per v1 behavior.

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

jfrazelle committed:
Merge pull request #13597 from sanketsaurav/master

Proposing Bose, Raman and Ramanujan

jfrazelle committed:
Merge pull request #13536 from Mic92/master

remove redundant mount/unmount calls on commit

jfrazelle committed:
Merge pull request #13611 from duglin/Issue13417a

Allow .dockerignore to ignore everything

Commit:
Merge pull request #13609 from Microsoft/10662-refixserverwindows

Windows: Fix windows to match linux after compile break

Commit:
Merge pull request #13644 from mchiang0610/patch-1

Fix - Download button does not download

Commit:
Networking docs: add reference to ndppd

Signed-off-by: Ian Bishop <ianbishop@pace7.com>

tiborvass committed:
Merge pull request #13651 from RichardScothern/v1-mirror-fix

Fallback to pull from hub if mirrors unavailable.

lizf-os committed:
test: Skip TestDevicePermissions on lxc

Closes: #13641

Signed-off-by: Zefan Li <lizefan@huawei.com>

chriswahl committed:
Corrected VMWare to VMware

Corrected spelling of VMware.

Signed-off-by: Chris Wahl <github@wahlnetwork.com>

vdemeester committed:
Add test coverage for pkg/tarsum

And removing unused code.
- tarsum.go :
  NewTarSumHash could be non exported (for now)
  NewTarSumForLabel is never used, except for the tests
- fileinfosums.go:
  SortByPos is never used, except for the tests
- versionning.go:
  GetVersions is never used, expect for the tests

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

kvasdopil committed:
update go-zfs to last version

Signed-off-by: Alexey Guskov <lexag@mail.ru>

jamtur01 committed:
Merge pull request #13657 from chriswahl/patch-1

Corrected VMWare to VMware

vdemeester committed:
Add test coverage to pkg/jsonmessage

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

runcom committed:
Remove old struct

Signed-off-by: Antonio Murdaca <antonio.murdaca@gmail.com>

estesp committed:
Merge pull request #13553 from Microsoft/10662-revert908db

Windows: Undo 908db518 for Windows daemon

runcom committed:
Fix wrong kill signal parsing

Signed-off-by: Antonio Murdaca <antonio.murdaca@gmail.com>

lizf-os committed:
Cleanup Daemon.verifyVolumesInfo() a bit

vols.VolumesRW has been initialized so it can't be nil. Furthermore
it's ok to read a nil map.

Signed-off-by: Zefan Li <lizefan@huawei.com>

vbatts committed:
Merge pull request #13630 from vdemeester/pkg-tarsum-test-coverage

Add test coverage for pkg/tarsum

thaJeztah committed:
Merge pull request #13633 from ZJU-SEL/fix-doc

fix typo.

LK4D4 committed:
Merge pull request #13664 from runcom/dead-code

Remove old struct

Commit:
Merge pull request #13666 from runcom/13665-kill-signal-wrong-handled

Fix wrong kill signal parsing

thaJeztah committed:
Merge pull request #13654 from porjo/netipv6

Networking docs: add reference to ndppd

calavera committed:
Merge pull request #13607 from vieux/PrintfIfNotEmpty

do not print empty values in docker info

calavera committed:
Bump libnetwork for 1.7 release.

Signed-off-by: David Calavera <david.calavera@gmail.com>

jfrazelle committed:
Merge pull request #13656 from lizf-os/13641-skip-test-on-lxc

test: Skip TestDevicePermissions on lxc

Commit:
Merge pull request #13576 from stevvooe/verify-digests

Properly verify manifests and layer digests on pull

estesp committed:
Merge pull request #13650 from jvgogh/master

Upon HTTP 302 redirect do not include "Authorization" header on 'untr…

calavera committed:
Update libnetwork to 4ded6fe3641b71863cc5985652930ce40efc3af4

Signed-off-by: David Calavera <david.calavera@gmail.com>

estesp committed:
Merge pull request #13668 from calavera/bump_libnetwork

Bump libnetwork for 1.7 release.

LK4D4 committed:
Add syslog-tag option for syslog log-driver

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Change address to syslog-address in syslog docs

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Tweaking text identified in PR #13147

Signed-off-by: Mary Anthony <mary@docker.com>

LK4D4 committed:
Wrap lines for journald docs

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

thaJeztah committed:
Merge pull request #13671 from moxiegirl/carry-13147-fix

Tweaking text identified in PR #13147

duglin committed:
Merge pull request #13661 from vdemeester/pkg-jsonmessage-test-coverage

Add test coverage to pkg/jsonmessage

LK4D4 committed:
Merge pull request #13591 from kvasdopil/update-go-zfs

Update go-zfs to last version

calavera committed:
Merge pull request #12838 from fntlnz/test-cmd

Using dockerCmd when possible

Commit:
Expose old config field for api < 1.19

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

calavera committed:
Merge pull request #13675 from runcom/13663-config-regression

Expose old config field for api < 1.19

Commit:
Update vendoring

Update libnetwork to 005bc475ee49a36ef2ad9c112d1b5ccdaba277d4
Synchronize changes to distribution (was missing _test.go file)

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

jfrazelle committed:
Merge pull request #13679 from icecrime/update_vendoring

Update vendoring

crosbymichael committed:
Fix nat integration tests

This removes complexity of current implementation and makes the test
correct and assert the right things.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

jfrazelle committed:
Merge pull request #13683 from crosbymichael/nat-test-fixes

Fix nat integration tests

hqhq committed:
Add docs for ip-forward

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq committed:
Add comment about inspect output

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

runcom committed:
Merge pull request #13696 from hqhq/hq_add_comment_inspect

Add comment about inspect output

Commit:
Merge pull request #13205 from hqhq/hq_fix_ipforward_doc

Add docs for ip-forward

thaJeztah committed:
Merge pull request #13672 from LK4D4/syslog_tag

Add syslog-tag option

runcom committed:
Do not omit empty json field in /containers/json api response

Signed-off-by: Antonio Murdaca <runcom@linux.com>

BenSeverson committed:
quick doc fix for windows versions

Signed-off-by: Ben Severson <BenSeverson@users.noreply.github.com>

LK4D4 committed:
Merge pull request #13231 from Microsoft/10662-chrootarchive

Windows: chrootarchive refactor

tianon committed:
Swap build-* to use UTC instead of local time

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

LK4D4 committed:
Support CloseNotifier for events

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

calavera committed:
Merge pull request #13645 from calavera/fix_stats_flaky_test

Fix stats flaky test.

jessfraz committed:
Update urls from .com to .org.

I added 301 redirects from dockerproject.com to dockerproject.org but may as
well make sure everything is updated anyways.

Signed-off-by: Jessica Frazelle <princess@docker.com>

thaJeztah committed:
Merge pull request #13702 from BenSeverson/patch-1

quick doc fix for windows versions

Commit:
Merge pull request #13703 from runcom/13691-container-ps-regression

Do not omit empty json field in /containers/json api response

runcom committed:
Merge pull request #13165 from ahmetalpbalkan/durations

Allow duration strings as --since/--until

LK4D4 committed:
Update github.com/Sirupsen/logrus to 0.8.2

Fix #13708

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Moving experimental

Signed-off-by: Mary Anthony <mary@docker.com>

Commit:
Merge pull request #13712 from LK4D4/fix_leak_events

Support CloseNotifier for events

runcom committed:
SizeRW & SizeRootFs omitted if empty in /container/json call

Signed-off-by: Antonio Murdaca <runcom@linux.com>

jfrazelle committed:
Merge pull request #13709 from jfrazelle/update-url

Update urls from .com to .org.

Commit:
Merge pull request #13715 from LK4D4/update_logrus

Update github.com/Sirupsen/logrus to 0.8.2

calavera committed:
Migrate data from old vfs paths to new local volumes path.

Signed-off-by: David Calavera <david.calavera@gmail.com>

jessfraz committed:
fix version struct on old versions

Signed-off-by: Jessica Frazelle <princess@docker.com>

jfrazelle committed:
Merge pull request #13718 from runcom/fix-containers-json

SizeRW & SizeRootFs omitted if empty in /container/json call

Commit:
Merge pull request #13731 from jfrazelle/fix-version-on-old-versions

fix version struct on old versions

mavenugo committed:
Using container NetworkDisabled to fix #13725

container.config.NetworkDisabled is set for both daemon's
DisableNetwork and --networking=false case. Hence using
this flag instead to fix #13725.

There is an existing integration-test to catch this issue,
but it is working for the wrong reasons.

Signed-off-by: Madhu Venugopal <madhu@docker.com>

Commit:
Merge pull request #13732 from mavenugo/netdisabled

Using container NetworkDisabled to fix #13725

jfrazelle committed:
Merge pull request #13704 from tianon/build-utc

Swap build-* to use UTC instead of local time

lizf-os committed:
Don't forget to clenaup tmpdir in TestFileSpecPlugin()

Also remove redundant code.

Signed-off-by: Zefan Li <lizefan@huawei.com>

aboch committed:
Fix for #13720

Signed-off-by: Alessandro Boch <aboch@docker.com>

SvenDowideit committed:
Tiny spelling issue in the debian control file :)

Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

Commit:
Fix kr/pty import path

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Merge pull request #13716 from moxiegirl/move-experimental

Moving experimental

calavera committed:
Merge pull request #13733 from lizf-os/discovery-test-remove-tmpdir

Don't forget to clenaup tmpdir in TestFileSpecPlugin()

LK4D4 committed:
Merge pull request #13586 from kvasdopil/zfs-88chars

Avoid 88-chars mountpoint length limit on freebsd

Commit:
Rename EXPERIMENTAL.md to README.md

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Remove reference to experimental release

Remove reference to experimental releases as it is really a nightly
channel rather than a scheduled release.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

LK4D4 committed:
Merge pull request #13745 from icecrime/test_import_path

Fix kr/pty import path in test-integration-cli

tianon committed:
Merge pull request #13736 from SvenDowideit/double-the-double-the-lines

Tiny spelling issue in the debian control file :)

crosbymichael committed:
Merge pull request #13728 from calavera/copy_old_volume_content

Migrate data from old vfs paths to new local volumes path.

Commit:
Merge pull request #13734 from aboch/master

Fix for #13720. Unpublished ports missing in docker ps

cpuguy83 committed:
Fix goroutine leak on logs -f with no output

Also noticed potential hang when only stdout or stderr are used with
follow=1

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cpuguy83 committed:
Merge pull request #13684 from lizf-os/cleanup-verify-volumes-info

Cleanup Daemon.verifyVolumesInfo() a bit

runcom committed:
Merge pull request #13648 from tiborvass/print-debugtransport-to-test-output

registry: debugTransport should print with testing.T.Log

tianon committed:
Fix release script to release _both_ .deb files

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

calavera committed:
Merge pull request #13640 from Microsoft/10662-implementmeminfo

Windows: Implement ReadMemInfo()

jfrazelle committed:
Merge pull request #13752 from tianon/fix-release-debs

Fix release script to release _both_ .deb files

SvenDowideit committed:
Bring over DHE docs updates for publishing

Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

SvenDowideit committed:
Merge pull request #13760 from SvenDowideit/update-dhe-docs-jun-5

Bring over DHE docs updates for publishing

estesp committed:
Merge pull request #13751 from cpuguy83/use_closenotifier_for_logs

Fix goroutine leak on logs -f with no output

jessfraz committed:
fix lxc build

Signed-off-by: Jessica Frazelle <princess@docker.com>

Commit:
Remove duplicate call to net.ParseIP
and a little cleanup

Signed-off-by: Doug Davis <dug@us.ibm.com>

thaJeztah committed:
Merge pull request #13748 from icecrime/experimental_docs

Minor changes to experimental docs

Commit:
Merge pull request #13636 from tiborvass/refactor-tls

Refactor TLS code with a new `tlsconfig` package

LK4D4 committed:
Merge pull request #13768 from jfrazelle/fix-lxc-again-bleh

fix lxc build

jfrazelle committed:
Merge pull request #13767 from duglin/TweakInNetOpts

Remove duplicate call to net.ParseIP

jfrazelle commented:
Closing because it has been 7 days without updates and there is something wrong with this branch, please if you want to make the patch open a new PR
